libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1939,19 +1941,11 @@ void WaveTrack::ApplyStretchRatio(
                  GetEndTime();
    if (startTime >= endTime)
       return;
-   const auto numChannels = NChannels();
-   auto count = 0;
-   const ProgressReporter reportChannelProgress =
-      reportProgress ?
-         [&](double progress) {
-            reportProgress((count + progress) / numChannels);
-         } :
-         ProgressReporter {};
-   for (const auto pChannel : TrackList::Channels(this))
-   {
-      pChannel->ApplyStretchRatioOne(startTime, endTime, reportChannelProgress);
-      ++count;
-   }
+   BasicUI::SplitProgress<WaveTrack*>(
+      reportProgress, TrackList::Channels(this),
+      [&](WaveTrack* pChannel, ProgressReporter child) {
+         pChannel->ApplyStretchRatioOne(startTime, endTime, child);
+      });
```

Comment:
This commit reuses the utility, for nested loops.

Maybe I understand better now why that led to the strange inverted flow of control.

Still I don't like it.  The simpler alternative to eliminate repetition would be to write another utility that wraps one progress indicator in another, that takes another denominator and another reference to a count.

Reply:
Yes like that BUT — the lambda must capture with = not & or you get a dangling reference!

Reply:
Following-up our discussion, I tried your idea to get inspiration from `std::for_each` - _that_ is the kind of suggestion I was hoping you were going to make from the beginning. Works like a charm, see [Morre standard-conform SplitProgress](https://github.com/audacity/audacity/pull/5043/commits/3d0ba5ae52150a11564272ab476aec01bc4c937d).